bounce.py

```
@@ -23,21 +22,82 @@
+
+    if not validators.domain(domain):
+        return ""
+
+    return domain
+
 
 @app.route('/')
 def bounce():
+    client_ip = get_client_ip()
+    if is_rate_limited(client_ip):
+        logging.warning(f"Rate limit exceeded for IP: {client_ip}")
```

Ruff

Logging statement uses f-string

(G004)